go-controller/pkg/ovn/egressqos.go

```diff
@@ -344,9 +348,20 @@ func (oc *DefaultNetworkController) processNextEgressQoSWorkItem(wg *sync.WaitGr
 
 	defer oc.egressQoSQueue.Done(key)
 
-	err := oc.syncEgressQoS(key.(string))
+	eqKey := key.(string)
+	eq, err := oc.getEgressQoS(eqKey)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to retrieve %s qos object: %v", eqKey, err))
+		oc.egressQoSQueue.Forget(key)
+		return true
+	}
+
+	err = oc.syncEgressQoS(eqKey, eq)
 	if err == nil {
 		oc.egressQoSQueue.Forget(key)
+		if err = oc.setEgressQoSStatus(eq, nil); err != nil {
```

Comment:
ohhhh wait, is the design such that every zone is actually updating the qos status and then the final "string" is set by the CM if all the statuses are error-free?
I really want to see a sample output where things look good on IC and things look bad on IC as to how the final status across each zone looks like in this model (again wish we had updated the enhancement first so that my q's around design would have been answered already)

Reply:
if so the logic will be dumb in nonIC because there its just the master process doing this in one go right?
I want to see how that looks like on nonIC

Reply:
This method is for patching `status.Messages` which is at 'global' level for nonIC case.